### PR TITLE
Tests for rwsem & percpu-rwsem

### PR DESCRIPTION
**Pull request by xairy**

### CODE REVIEW

File: mm/ktsan/stack.c [hunk 1]
```diff
@@ -14,7 +14,6 @@ void kt_stack_print(kt_stack_t *stack, uptr_t top_pc)
 		pc = kt_decompress(stack->pc[i]);
 		pr_err(" [<%p>] %pS\n", (void *)pc, (void *)pc);
 	}
-	pr_err("\n");
```
Comment (dvyukov) [inline mm/ktsan/stack.c]: This was done on the purpose, all user-space sanitizers always terminate stacks with new line. This significantly improves report readability, and was suggested by external people who know how to make things look good.

Comment (xairy) [inline mm/ktsan/stack.c]: But in that case reports look like this:
```
==================================================================
ThreadSanitizer: data-race in rcu_read_under_lock

Read at 0xffff88028a19b400 of size 4 by thread 3052 on CPU 1:
 [<ffffffff81252e66>] rcu_read_under_lock+0x16/0x30 mm/ktsan/tests_inst.c:510
 [<ffffffff812523fd>] thr_func+0x2d/0x50 mm/ktsan/tests_inst.c:33
 [<ffffffff810b98c1>] kthread+0x161/0x180 kernel/kthread.c:209
 [<ffffffff81ea8c8f>] ret_from_fork+0x3f/0x70 arch/x86/entry/entry_64.S:529

Previous write at 0xffff88028a19b400 of size 4 by thread 3053 on CPU 2:
 [<ffffffff81252e36>] rcu_write_under_lock+0x16/0x30 mm/ktsan/tests_inst.c:523
 [<ffffffff812523fd>] thr_func+0x2d/0x50 mm/ktsan/tests_inst.c:33
 [<ffffffff810b98c1>] kthread+0x161/0x180 kernel/kthread.c:209
 [<ffffffff81ea8c8f>] ret_from_fork+0x3f/0x70 arch/x86/entry/entry_64.S:529

==================================================================
```

And that trailing newline before ====== doesn't look good, imho.

All other kt_stack_print are followed by pr_err("\n") in report.c

Comment (dvyukov) [inline mm/ktsan/stack.c]: > All other kt_stack_print are followed by pr_err("\n") in report.c

We need to remove them.

> And that trailing newline before ====== doesn't look good, imho.

It does not look good, but it is better than missed newlines and double newlines if manually sprinkle pr_err("\n") throughout the code.
We can add a newline before ThreadSanitizer: data-race for consistency.

Comment (xairy) [inline mm/ktsan/stack.c]: Dropped commit.

**Comment by dvyukov** [comment]
LGTM

**Pull request merged**